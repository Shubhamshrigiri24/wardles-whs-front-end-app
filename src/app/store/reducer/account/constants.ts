export const ACCOUNT_ACTIONS = {
  SET_ACCOUNT: "SET_ACCOUNT",
  GET_ACCOUNT: "GET_ACCOUNT",
  SET_ACCOUNT_STATUS: "SET_ACCOUNT_STATUS",
};
