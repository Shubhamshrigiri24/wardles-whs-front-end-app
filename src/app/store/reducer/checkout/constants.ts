import {
  CustomerDetails,
  DeliveryDetails,
} from "app/store/reducer/checkout/types";

export const ADD_CUSTOMER_DETAILS = "CUSTOMER/ADD_CUSTOMER_DETAILS";
export const ADD_DELIVERY_DETAILS = "CUSTOMER/ADD_DELIVERY_DETAILS";

export const FETCH_BRAINTREE_TOKEN = "PAYMENT/FETCH_BRAINTREE_TOKEN";
export const FETCH_BRAINTREE_TOKEN_REQUEST =
  "PAYMENT/FETCH_BRAINTREE_TOKEN_REQUEST";
export const FETCH_BRAINTREE_TOKEN_RESPONSE =
  "PAYMENT/FETCH_BRAINTREE_TOKEN_RESPONSE";
export const FETCH_BRAINTREE_TOKEN_FAILURE =
  "PAYMENT/FETCH_BRAINTREE_TOKEN_FAILURE";

export const FETCH_ADDRESS_FROM_POSTCODE =
  "ADDRESS/FETCH_ADDRESS_FROM_POSTCODE";
export const FETCH_ADDRESS_FROM_POSTCODE_REQUEST =
  "ADDRESS/FETCH_ADDRESS_FROM_POSTCODE_REQUEST";
export const FETCH_ADDRESS_FROM_POSTCODE_RESPONSE =
  "ADDRESS/FETCH_ADDRESS_FROM_POSTCODE_RESPONSE";
export const FETCH_ADDRESS_FROM_POSTCODE_FAILURE =
  "ADDRESS/FETCH_ADDRESS_FROM_POSTCODE_FAILURE";

export const emptyCustomerDetails: CustomerDetails = {
  firstName: "",
  lastName: "",
  line1: "",
  line2: "",
  city: "",
  postcode: "",
  email: "",
  phone: "",
  surgeryName: "",
  surgeryAddress: "",
  allowMarketing: false,
};

export const emptyDeliveryDetails: DeliveryDetails = {
  title: "",
  firstName: "",
  lastName: "",
  line1: "",
  line2: "",
  city: "",
  postcode: "",
};
