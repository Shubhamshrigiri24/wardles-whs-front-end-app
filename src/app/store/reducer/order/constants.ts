export const ORDER_ACTIONS = {
  SET_CUSTOMER: "SET_CUSTOMER",
  SET_PRODUCT: "SET_PRODUCT",
  SET_PAYMENT: "SET_PAYMENT",
  CLEANUP_ORDER_FLOW_STATE: "CLEANUP_ORDER_FLOW_STATE",
};

export const CUSTOMER_TYPES = {
  GUEST: "GUEST",
  EXISTING: "EXISTING",
  NEW: "NEW",
  PENDING: "PENDING", // Used to immediately trigger registration when redirected from sign-in PIN page
};
