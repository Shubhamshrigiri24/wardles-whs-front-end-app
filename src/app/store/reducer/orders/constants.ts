export const ORDERS_ACTIONS = {
  SET_ORDER: "SET_ORDER",
  SET_ORDERS: "SET_ORDERS",
  GET_ORDERS: "GET_ORDERS",
  SET_ORDERS_STATUS: "SET_ORDERS_STATUS",
};
