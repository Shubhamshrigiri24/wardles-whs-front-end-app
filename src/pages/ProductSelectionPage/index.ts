import { connect } from "react-redux";
import { RootState } from "../../app/store/types";
import ProductSelectionPage, {
  ProductSelectionStateProps,
  ProductSelectionDispatchProps,
} from "./ProductSelectionPage";
import {
  setProduct,
  setPack,
  fetchProducts,
  setOnlineService,
  checkProductToken,
  setProductToken,
  makeSetConsultation,
} from "../../app/store/reducer/online/actions";
import {
  getProducts,
  getProductsFetching,
  getOnlineServices,
  makeGetEligiblePacks,
  getSelectedOnlineService,
  getProductTokenValid,
  getProductToken,
} from "../../app/store/selectors";
import {
  getConsultationResponses,
  getSelectedProduct,
} from "app/store/reducer/online/selectors";
import { Action, Dispatch } from "redux";

function mapStateToProps(state: RootState): ProductSelectionStateProps {
  return {
    products: getProducts(state),
    productToken: getProductToken(state),
    productTokenValid: getProductTokenValid(state),
    selectedProduct: getSelectedProduct(state.online),
    getEligiblePacks: makeGetEligiblePacks(state),
    fetching: getProductsFetching(state),
    services: getOnlineServices(state),
    selectedService: getSelectedOnlineService(state),
    consultation: getConsultationResponses(state.online),
  };
}

const mapDispatchToProps: (
  dispatch: Dispatch<Action>
) => ProductSelectionDispatchProps = (dispatch: Dispatch<Action>) => ({
  setPack: (...args) => dispatch(setPack(...args)),
  fetchProducts: (...args) => dispatch(fetchProducts(...args)),
  setProductToken: (...args) => dispatch(setProductToken(...args)),
  checkProductToken: (...args) => dispatch(checkProductToken(...args)),
  setProduct: (...args) => dispatch(setProduct(...args)),
  setOnlineService: (...args) => dispatch(setOnlineService(...args)),
  setConsultation: makeSetConsultation(dispatch),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ProductSelectionPage);
