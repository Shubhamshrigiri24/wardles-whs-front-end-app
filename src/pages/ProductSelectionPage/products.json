[
  {
    "label": "4 pack",
    "quantity": 4,
    "price": 19.99,
    "value": 5
  },
  {
    "label": "8 pack",
    "quantity": 8,
    "price": 34.96,
    "value": 4.38
  },
  {
    "label": "12 pack",
    "quantity": 12,
    "price": 52,
    "value": 4.33
  },
  {
    "label": "16 pack",
    "quantity": 16,
    "price": 52,
    "value": 4.06,
    "isBestValue": true
  }
]
