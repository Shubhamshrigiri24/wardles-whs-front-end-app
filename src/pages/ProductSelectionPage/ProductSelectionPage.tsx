import React, { useEffect, useState } from "react";
import { Box, Container } from "@material-ui/core";
import { useHistory, useParams } from "react-router-dom";
import { analytics } from "@welldigital/ui-common/Analytics";
import USPBanner from "../../components/USPBanner";
import {
  defaultOnlineService,
  defaultPack,
  OnlineQuestion,
  OnlineService,
  OnlineServiceIds,
  OnlineServiceParamTypes,
  Pack,
  Product,
  ProductSelectionPageQuery,
  ServiceTextValue,
  SetOnlineServiceAction,
  SetPackAction,
  SetProductAction,
  TextByServiceMap,
} from "app/store/reducer/online/types";
import {
  Button,
  Product as ProductDetails,
  ProductPack,
  Text,
} from "../../components";
import { useLocationQuery } from "../../utils/utils";
import { isHana1315Flow } from "../../utils/onlineServiceConsultation";
import { HanaTripettoQuestions } from "../../constants/hanaTripettoQuestions";
import { EdProductIds, edProductsMap } from "../../components/EdProducts";

export interface ProductSelectionStateProps {
  products: Product[];
  productToken: string | null;
  productTokenValid: boolean;
  selectedProduct: Product;
  getEligiblePacks: (product: Product) => Pack[];
  fetching: boolean;
  services: OnlineService[];
  selectedService: OnlineService;
  consultation: OnlineQuestion[];
}

export interface ProductSelectionDispatchProps {
  setPack: (data: Pack) => SetPackAction;
  fetchProducts: (onlineServiceId: string) => void;
  setProductToken: (token: string) => void;
  checkProductToken: (token: string) => void;
  setProduct: (data: Product) => SetProductAction;
  setOnlineService: (id: OnlineService) => SetOnlineServiceAction;
  setConsultation: (service: string, consultation: OnlineQuestion[]) => void;
}

export type ProductSelectionProps = ProductSelectionStateProps &
  ProductSelectionDispatchProps;

const ProductSelectionPage: React.FC<ProductSelectionProps> = ({
  setPack,
  setProduct,
  selectedProduct,
  products,
  productTokenValid,
  consultation,
  fetchProducts,
  setProductToken,
  checkProductToken,
  fetching,
  setOnlineService,
  services,
  selectedService,
  getEligiblePacks,
  setConsultation,
}) => {
  const history = useHistory();
  const { token } = useLocationQuery<ProductSelectionPageQuery>();
  const { onlineServiceId, productId } = useParams<OnlineServiceParamTypes>();
  const [productPacks, setProductPacks] = useState([defaultPack]);
  const [selectedPackSku, setPackSku] = useState<string | null>(null);

  useEffect(() => {
    analytics.trackEvent({
      flow: onlineServiceId,
      event: `View product page`,
    });
  }, [onlineServiceId]);

  useEffect(() => {
    if (!token || onlineServiceId !== OnlineServiceIds.hana) return;
    checkProductToken(token);
  }, [token, checkProductToken, onlineServiceId]);

  useEffect(() => {
    if (!token || onlineServiceId !== OnlineServiceIds.hana) return;
    if (productTokenValid) {
      setProductToken(token);
      // if the one-time link is opened in another browser, we won't have any localStorage data,
      // we need this to be recognized in confirmation page, for showing additional info section
      if (!isHana1315Flow(consultation)) {
        setConsultation(onlineServiceId, [
          {
            number: 0,
            question: HanaTripettoQuestions.q1,
            answer: "14",
            type: "int",
          },
        ]);
      }
    } else {
      history.push("/page-not-found");
    }
  }, [
    token,
    productTokenValid,
    setProductToken,
    consultation,
    setConsultation,
    history,
    onlineServiceId,
  ]);

  useEffect(() => {
    fetchProducts(onlineServiceId);
    const selectedOnlineService =
      services.find(({ id }) => id === onlineServiceId) || defaultOnlineService;
    if (selectedOnlineService !== selectedService) {
      setOnlineService(selectedOnlineService);
    }
  }, [
    setOnlineService,
    fetchProducts,
    onlineServiceId,
    services,
    selectedService,
  ]);

  useEffect(() => {
    if (!fetching && products.length > 0) {
      let product;
      if (onlineServiceId === OnlineServiceIds.ed) {
        product = products.find(({ id }) => id === productId);
      } else {
        product = products[0];
      }
      if (!product) return;
      setProduct(product);
      setProductPacks(getEligiblePacks(product));
    }
  }, [
    fetching,
    products,
    setProductPacks,
    setProduct,
    getEligiblePacks,
    onlineServiceId,
    productId,
  ]);

  const continueToCheckout = () => {
    if (!selectedPackSku) return;
    const selectedPack = productPacks.find(
      ({ sku }) => selectedPackSku === sku
    );
    if (!selectedPack) return;
    setPack(selectedPack);
    analytics.trackEvent({
      flow: onlineServiceId,
      event: "Selected pack size",
      metadata: {
        product: selectedPack,
      },
    });
    history.push(`/online/${onlineServiceId}/checkout/your-details`);
  };

  const isEd = onlineServiceId === OnlineServiceIds.ed;
  const productText: ServiceTextValue = TextByServiceMap[onlineServiceId];
  return (
    <Container maxWidth={"xs"} disableGutters>
      <Text variant={"beforeTitle"}>{productText.beforeTitle}</Text>
      <Text variant={"title"}>{productText.title}</Text>
      <Text variant={"paragraph"}>{productText.info1}</Text>
      {isEd ? (
        <Box marginBottom={5}>
          <ProductDetails
            {...edProductsMap[productId as unknown as EdProductIds]}
          />
        </Box>
      ) : (
        <>
          <Text variant={"label"}>{selectedProduct.name}</Text>
          {!!productText.info2 && (
            <Text variant={"paragraph"}>{productText.info2}</Text>
          )}
          <USPBanner product={selectedProduct} />
        </>
      )}

      <Box marginTop={4} marginBottom={3}>
        <Text variant={"label"}>Select a pack size</Text>
        {fetching ? (
          <div>Loading...</div>
        ) : (
          productPacks.map((pack, k) => (
            <ProductPack
              key={k}
              {...pack}
              checked={pack.sku === selectedPackSku}
              onCheck={() => setPackSku(pack.sku)}
            />
          ))
        )}
      </Box>

      {isEd && <USPBanner product={selectedProduct} />}

      <Box marginTop={5}>
        <Button
          data-testid={"ProductSelectionPage/continue-button"}
          variant={"formNext"}
          onClick={continueToCheckout}
          disabled={!selectedPackSku}
          children={"Continue to checkout"}
        />
      </Box>
    </Container>
  );
};

export default ProductSelectionPage;
