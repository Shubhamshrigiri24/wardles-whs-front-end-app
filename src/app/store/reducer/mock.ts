import faker from "faker/locale/en_GB";
import { CustomerDetails, DeliveryDetails } from "./checkout/types";
import { PostcodeLookupAddress } from "./shared";
import {
  OnlineQuestion,
  Pack,
  Product,
  SuccessfulConsultationResponse,
} from "./online/types";

export function makeCustomerDetails(
  override?: Partial<CustomerDetails>
): CustomerDetails {
  return {
    firstName: "firstName",
    lastName: "lastName",
    email: "a@b.com",
    postcode: "M19 2UL",
    line1: "line1",
    line2: "line2",
    city: "city",
    phone: "01189998819991197253",
    surgeryName: "surgeryName",
    surgeryAddress: "surgeryAddress",
    allowMarketing: true,
    ...override,
  };
}

export function makeDeliveryDetails(): DeliveryDetails {
  return {
    title: "Mrs",
    firstName: faker.name.firstName(),
    lastName: faker.name.lastName(),
    postcode: faker.address.zipCode(),
    line1: faker.address.streetName(),
    line2: faker.address.streetName(),
    city: faker.address.city(),
  };
}

export function makePostCodeLookUpAddresses(): PostcodeLookupAddress[] {
  return [
    {
      line1: "64 Broom Avenue",
      line2: "",
      city: "Manchester",
      country: "England",
      postcode: "M19 2UL",
    },
    {
      line1: "66 Broom Avenue",
      line2: "",
      city: "Manchester",
      country: "England",
      postcode: "M19 2UL",
    },
    {
      line1: "68 Broom Avenue",
      line2: "",
      city: "Manchester",
      country: "England",
      postcode: "M19 2UL",
    },
  ];
}

export function makeSuccessfulConsultationResponse(
  id: string = "default",
  override?: Partial<SuccessfulConsultationResponse>
): SuccessfulConsultationResponse {
  return {
    answer: `answer-${id}`,
    answerString: `answerString-${id}`,
    initialQuestion: `question-${id}`,
    kind: `trepetto-input-type-${id}`,
    question: `question-${id}`,
    type: "string",
    ...override,
  };
}

export function makeOnlineQuestion(number: number = 1): OnlineQuestion {
  return {
    number,
    question: `question-${number}`,
    answer: `answer-${number}`,
    type: "string",
  };
}

export function makeOnlineQuestions(): OnlineQuestion[] {
  return [makeOnlineQuestion(1), makeOnlineQuestion(2), makeOnlineQuestion(3)];
}

export function makeProduct(
  id: string = "default",
  override?: {
    id?: string;
    name?: string;
    service?: string;
    packs?: Pack[];
    variant?: string;
  }
): Product {
  return {
    id,
    name: `name-${id}`,
    service: `service-${id}`,
    packs: [
      makePack(`${id}-pack-1`),
      makePack(`${id}-pack-2`),
      makePack(`${id}-pack-3`),
    ],
    variant: "",
    ...override,
  };
}

export function makePack(
  id: string = "default",
  override?: {
    sku?: string;
    name?: string;
    label?: string;
    price?: number;
    pricePerUnit?: number;
    itemsPerPack?: number;
  }
): Pack {
  return {
    sku: `sku-${id}`,
    name: `name-${id}`,
    label: `label-${id}`,
    basePrice: 1,
    price: 1,
    pricePerUnit: 1,
    itemsPerPack: 1,
    discounts: [],
    ...override,
  };
}

export const customerDetails: CustomerDetails = {
  firstName: "FIRST_NAME",
  lastName: "LAST_NAME",
  email: "EMAIL",
  postcode: "POSTCODE",
  line1: "LINE_1",
  line2: "LINE_2",
  city: "CITY",
  phone: "PHONE",
  surgeryName: "SURGERY_NAME",
  surgeryAddress: "SURGERY_ADDRESS",
  allowMarketing: true,
};

export const deliveryDetails: DeliveryDetails = {
  title: "Mrs",
  firstName: "FIRST_NAME",
  lastName: "LAST_NAME",
  postcode: "POSTCODE",
  line1: "LINE_1",
  line2: "LINE_2",
  city: "CITY",
};
