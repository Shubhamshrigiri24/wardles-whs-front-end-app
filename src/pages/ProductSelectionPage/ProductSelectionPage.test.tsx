import React from "react";
import { render, fireEvent, act } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import ProductSelectionPage, {
  ProductSelectionProps,
} from "./ProductSelectionPage";
import {
  Ed,
  Hana,
  OnlineProductIds,
  OnlineServiceIds,
} from "app/store/reducer/online/types";
import { makeProduct } from "app/store/reducer/mock";
import { HanaTripettoQuestions } from "../../constants/hanaTripettoQuestions";

let mock_token = "";
let mock_onlineServiceId: string;
let mock_productId: string | undefined;
const mock_historyPush = jest.fn();

jest.mock("react-router-dom", () => ({
  useParams: jest.fn(() => {
    return {
      onlineServiceId: mock_onlineServiceId,
      productId: mock_productId,
    };
  }),
  useHistory: () => ({
    push: mock_historyPush,
  }),
  useLocation: () => ({
    search: mock_token ? `?token=${mock_token}` : "",
  }),
}));

describe("<ProductSelectionPage />", () => {
  beforeEach(() => {
    mock_onlineServiceId = OnlineServiceIds.hana;
    mock_productId = undefined;
  });

  it("should match the snapshot", () => {
    const props = makeProps();
    const { container } = render(<ProductSelectionPage {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("should contain the sildenafil product selected in url", async () => {
    mock_onlineServiceId = OnlineServiceIds.ed;
    mock_productId = OnlineProductIds.sildenafil;
    const props = makeProps();
    const { getByTestId } = render(<ProductSelectionPage {...props} />);
    const sildenafilProduct = getByTestId("product-sildenafil");
    expect(sildenafilProduct).toBeTruthy();
  });

  it("should contain the viagra product selected in url", async () => {
    mock_onlineServiceId = OnlineServiceIds.ed;
    mock_productId = OnlineProductIds.viagra;
    const props = makeProps();
    const { getByTestId } = render(<ProductSelectionPage {...props} />);
    const sildenafilProduct = getByTestId("product-viagra");
    expect(sildenafilProduct).toBeTruthy();
  });

  it("should call fetch products on render", async () => {
    const props = makeProps();
    render(<ProductSelectionPage {...props} />);
    expect(props.fetchProducts).toBeCalledWith(props.selectedService.id);
  });

  it("should call set product when products are loaded", async () => {
    const props = makeProps();
    const { getByTestId } = render(<ProductSelectionPage {...props} />);
    await act(async () => {
      fireEvent.click(getByTestId("ProductSelectionPage/continue-button"));
    });
    expect(props.setProduct).toBeCalledWith(props.products[0]);
  });

  it("should not call check product token on render if not token present", async () => {
    const props = makeProps();
    render(<ProductSelectionPage {...props} />);
    expect(props.checkProductToken).not.toBeCalled();
  });

  it("should call check product token on render if token present in url", async () => {
    mock_token = "test-token";
    const props = makeProps();
    render(<ProductSelectionPage {...props} />);
    expect(props.checkProductToken).toBeCalledWith(mock_token);
    mock_token = "";
  });

  it("should call history push after if token is invalid", async () => {
    mock_token = "test-token";
    const props = makeProps({ productTokenValid: false });
    render(<ProductSelectionPage {...props} />);
    expect(mock_historyPush).toBeCalledWith("/page-not-found");
    expect(props.setProductToken).not.toBeCalled();
    expect(props.setConsultation).not.toBeCalled();
    mock_historyPush.mockClear();
    mock_token = "";
  });

  it("should not call history push after if token is valid, and set consultation if empty", async () => {
    mock_token = "test-token";
    const props = makeProps({ productTokenValid: true });
    render(<ProductSelectionPage {...props} />);
    expect(mock_historyPush).not.toBeCalled();
    expect(props.setProductToken).toBeCalledWith(mock_token);
    expect(props.setConsultation).toBeCalledWith(props.selectedService.id, [
      {
        type: "int",
        number: 0,
        question: HanaTripettoQuestions.q1,
        answer: "14",
      },
    ]);
    mock_historyPush.mockClear();
    mock_token = "";
  });

  it("should not call history push after if token is valid, and NOT set consultation if is already set", async () => {
    mock_token = "test-token";
    const props = makeProps({
      productTokenValid: true,
      consultation: [
        {
          type: "int",
          number: 0,
          question: HanaTripettoQuestions.q1,
          answer: "14",
        },
      ],
    });
    render(<ProductSelectionPage {...props} />);
    expect(mock_historyPush).not.toBeCalled();
    expect(props.setProductToken).toBeCalledWith(mock_token);
    expect(props.setConsultation).not.toBeCalled();
    mock_historyPush.mockClear();
    mock_token = "";
  });
});

function makeProps(p?: Partial<ProductSelectionProps>): ProductSelectionProps {
  const selectedProduct = makeProduct(mock_productId);
  return {
    setProduct: jest.fn(),
    services: [Ed, Hana],
    selectedService: mock_onlineServiceId === OnlineServiceIds.hana ? Hana : Ed,
    selectedProduct,
    productToken: p?.productToken ?? null,
    productTokenValid: p?.productTokenValid ?? true,
    products: [selectedProduct, selectedProduct],
    getEligiblePacks: (p) => p.packs,
    fetchProducts: jest.fn(),
    fetching: false,
    setPack: jest.fn(),
    setOnlineService: jest.fn(),
    setProductToken: jest.fn(),
    checkProductToken: jest.fn(),
    consultation: p?.consultation || [],
    setConsultation: jest.fn(),
  };
}
