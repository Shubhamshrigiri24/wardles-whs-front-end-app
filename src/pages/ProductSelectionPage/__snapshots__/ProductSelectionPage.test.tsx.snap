// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductSelectionPage /> should match the snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-disableGutters MuiContainer-maxWidthXs"
  >
    <div
      class="makeStyles-root-1 makeStyles-beforeTitle-3"
    >
      Contraception options
    </div>
    <div
      class="makeStyles-root-1 makeStyles-title-2"
    >
      Great, thank you!
      <br />
       Here are your contraception options
    </div>
    <div
      class="makeStyles-root-1 makeStyles-paragraph-4"
    >
      Thanks for answering those questions. You can choose your contraception from the options below, then complete checkout.
    </div>
    <div
      class="makeStyles-root-1 makeStyles-label-5"
    >
      name-default
    </div>
    <div
      class="makeStyles-root-7 makeStyles-infoType-8"
    >
      <div
        class="makeStyles-icon-9"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
          />
        </svg>
      </div>
      <div
        class="makeStyles-message-10"
      >
        Includes ongoing support from our pharmacists.
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-11"
    >
      <div
        class="makeStyles-root-1 makeStyles-label-5"
      >
        Select a pack size
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-12 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="makeStyles-inner-13"
        >
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-21 MuiRadio-root makeStyles-checkbox-18 MuiRadio-colorSecondary makeStyles-packLabel-16 MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-24"
                  type="radio"
                  value=""
                />
                <div
                  class="PrivateRadioButtonIcon-root-25"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-26"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              label-default-pack-1
            </span>
          </label>
          <div
            class="makeStyles-packLabelGreyContainer-15"
          >
            <p
              class="MuiTypography-root makeStyles-packLabelGrey-17 MuiTypography-body1"
            >
              £
              1.00
              (£1.00 per unit)
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-12 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="makeStyles-inner-13"
        >
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-21 MuiRadio-root makeStyles-checkbox-18 MuiRadio-colorSecondary makeStyles-packLabel-16 MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-24"
                  type="radio"
                  value=""
                />
                <div
                  class="PrivateRadioButtonIcon-root-25"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-26"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              label-default-pack-2
            </span>
          </label>
          <div
            class="makeStyles-packLabelGreyContainer-15"
          >
            <p
              class="MuiTypography-root makeStyles-packLabelGrey-17 MuiTypography-body1"
            >
              £
              1.00
              (£1.00 per unit)
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-12 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="makeStyles-inner-13"
        >
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-21 MuiRadio-root makeStyles-checkbox-18 MuiRadio-colorSecondary makeStyles-packLabel-16 MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-24"
                  type="radio"
                  value=""
                />
                <div
                  class="PrivateRadioButtonIcon-root-25"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-26"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              label-default-pack-3
            </span>
          </label>
          <div
            class="makeStyles-packLabelGreyContainer-15"
          >
            <p
              class="MuiTypography-root makeStyles-packLabelGrey-17 MuiTypography-body1"
            >
              £
              1.00
              (£1.00 per unit)
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-28"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-formNext-29 Mui-disabled Mui-disabled"
        data-testid="ProductSelectionPage/continue-button"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Continue to checkout
          <span
            class="MuiButton-endIcon MuiButton-iconSizeMedium"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </span>
        </span>
      </button>
    </div>
  </div>
</div>
`;
